server/api/auth/login.post.ts

```diff
@@ -4,25 +4,41 @@ import { z } from "zod";
 import { SESSION_COOKIE_NAME, SESSION_STORAGE, SESSION_TTL_SECONDS } from "~/utils/sessionStorage";
 import prisma from "~/utils/db";
 
+// Constants
+const STATUS_CODES = {
+  UNPROCESSABLE_ENTITY: 422,
+  UNAUTHORIZED: 401,
+};
+
+const ERROR_MESSAGES = {
+  VALIDATION_ERROR: "Username and password field is required",
+  USER_NOT_FOUND: "Invalid username or password",
+  INVALID_CREDENTIALS: "Invalid username or password",
+};
+
 const userSchema = z.object({
   username: z.string(),
   password: z.string(),
 });
 
 export default defineEventHandler(async (event) => {
+  // Validate request body
   const result = await readValidatedBody(event, body => userSchema.safeParse(body));
   if (!result.success) {
-    throw createError({ statusCode: 422, statusMessage: "Username and password field is required" });
+    throw createError({ statusCode: STATUS_CODES.UNPROCESSABLE_ENTITY, statusMessage: ERROR_MESSAGES.VALIDATION_ERROR });
   }
 
+  // Check for user existence
   const user = await prisma.user.findFirst({ where: { username: result.data.username } });
   if (!user) {
-    throw createError({ statusCode: 401, statusMessage: `Unauthorized` });
+    // User not found, send a generic message to prevent enumeration attacks
```

Comment:
This comment is too verbose and unnecessary because it's clear from the code what we do.
```suggestion
```